### PR TITLE
Store: Prompt user to confirm email before letting them save store address

### PR DESCRIPTION
**Pull request by allendav**
Partially addresses #20877 

This PR:
* adds a notice to the Building Your Store (plugins install) view if the user has not activated their WordPress.com account yet
* adds a notice to the Howdy! Ready to Start Selling (set store location) view if the user has not activated their WordPress.com account yet
* intercepts clicks on the "Let's Go" button on the Howdy! Ready to Start Selling (set store location) view if the user has not activated their WordPress.com account yet and pops a dialog for them to do so
* re-fetches the user (including the activation status) when they dismiss that dialog in case they completed activation in another tab so we don't continue to block "Let's Go" if they are now activated

To test:
* Open an incognito window
* Navigate to wordpress.com/start
* In step 1 of 4, give your site a name (e.g. “allendav201701041351” ) and tick the “Sell products or collect payments” checkbox
* Hit Continue.
* In step 2 of 4, select any theme by clicking on it
* In step 3 of 4, enter your site name again (e.g. “allendav201701041351” ) and then, after it has loaded, select the .blog domain name (e.g. allendav201701041351.blog)
* In step 4 of 5, select the business plan
* In step 4 of 5, leave your username as is (e.g. allendav201701041351), give a unique email address (e.g. allendavidsnook+1351@gmail.com) and choose a password
* Wait for the “give is a minute to finish”
* Wait for the privacy protection and domain contact information view to load
* Open another window as an automattician and give your new user 300 free credits
* Return to the privacy protection and domain contact information view and then Continue to Checkout
* In the Secure Payment window, click on “Pay with Credits”
* Wait for it to complete your purchase
* On the “Thank you for your purchase screen, verify you have the “We’ve sent a message…confirm your address” notice at the top
* Click on “Create Your Store”
* Wait while the “Building Your Store” view loads and completes
* You’ll be dropped on the “Howdy! Ready to start selling? First we need to know where you are in the world.” view.
* Stop.
* In the same incognito window, switch from wordpress.com to this branch - e.g. navigate to http://calypso.localhost:3000/store/allendav201701041351.blog
* Make sure you get the “You need to confirm…” notice at the top (above the address form)
* Enter your address (it should be pre-filled in from your domain contact information) and hit “Let’s Go”
* Make sure that you immediately get the “Please verify your email address” dialog
* Hit OK to close the dialog
* Try “Let’s Go!” again
* Make sure that you immediately get the “Please verify your email address” dialog again
* Go open your email, find the “Activate allendav201701041351” email, and “Copy Link Address” from Confirm Now button therein. Paste it into a new tab in the incognito window.
* Close that tab after it is done processing
* Return to the tab with the “Howdy! Ready to start selling? First we need to know where you are in the world.” view
* Hit OK to close the Please verify your email address dialog
* Try “Let’s Go!” again
* Verify you are delivered to the “Let's set up your store. Here are the things you’ll need to do” task list view

### CODE REVIEW

File: client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js [hunk 1]
```diff
@@ -425,16 +429,19 @@ class RequiredPluginsInstallView extends Component {
...
-					<ProgressBar value={ progress } total={ totalSeconds } isPulsing />
-				</SetupHeader>
+			<div className="dashboard__setup-wrapper">
+				<SetupNotices />
+				<div className="card dashboard__plugins-install-view">
+					{ site && <QueryJetpackPlugins siteIds={ [ site.ID ] } /> }
+					<SetupHeader
+						imageSource={ '/calypso/images/extensions/woocommerce/woocommerce-store-creation.svg' }
+						imageWidth={ 160 }
+						title={ translate( 'Building your store' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js]: Hi! I've found a possible matching string that has already been translated **4** times:
`translate( 'Get started building your new:' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by justinshreve** [comment]
Looks like a canary might have timed out by the way.

**Comment by timmyc** [comment]
One thing to note with this PR *all* new users who go through store signup will have to now verify their account ( if not already done ) before being able to proceed past the address step. At first I thought we should consider only requiring this for non-supported countries, but since we still direct "supported country" users to the customizer in the setup checklist - which is in wp-admin and requires the account to be verified - i think it is good to require it for all users.

Just wanted to make sure everyone was aware that is new verification requirement will impact all new accounts going through the store signup /cc @kellychoffman

**Comment by allendav** [comment]
@timmyc wrote

> The first ( most recent one ) has the word Verify in the subject line, but it is the domain reg verification where the one needed the subject line is for "Activate %username%"

Yeah, I'm not a fan of that either. I've written that up as https://github.com/Automattic/wp-calypso/issues/21295

**Comment by allendav** [comment]
@justinshreve wrote

> Looks like a canary might have timed out by the way.

Yeah, its not related to these changes, and restarting it isn't helping. Will try a little bit more before ignoring it.

**Pull request merged**